### PR TITLE
[Bug]: Fix domain name validation

### LINKED ISSUE #17667: [Bug]: Using an environment variable for "pimcore.general.domain" is not possible anymore
### Pimcore version

11.4.0

### Steps to reproduce

Adapt the config.yaml and add 
```
pimcore:
  general:
    domain: '%env(APP_URL)%'
```

### Actual Behavior

Open Pimcore in the browser. You will get the exception:
```
Invalid configuration for path "pimcore.general.domain": Invalid domain name ""%env(APP_URL)%""
```

### Expected Behavior

The url becomes processed correctly (before validation)
This worked before https://github.com/pimcore/pimcore/pull/17221

### LINKED ISSUE #17669: [Bug]: Wildcard Domains for the additional Domains in Sites not more possible.
### Pimcore version

11.4.0

### Steps to reproduce

Add a Wildcard Domain in the "Additional Domains" Field in the Sites Formular. 

Bildschirmfoto 2024-10-01 um 16 59 04

The Problem is the "FILTER_VALIDATE_DOMAIN" Flag in the setDomains Function from the Sites Model. 

Bildschirmfoto 2024-10-01 um 17 07 48

The "FILTER_VALIDATE_DOMAIN" checks the RFC-compliant structure of a domain name. A wildcard (*) is not a valid part of a host name because, according to RFC standards, host names may not begin with special characters such as *.

### Actual Behavior

The System throw an exception.

### Expected Behavior

The Wildcard Domain is stored.

### PR DESCRIPTION
**Pull request by kingjia90**
## Changes in this pull request  
Resolves https://github.com/pimcore/pimcore/issues/17667, https://github.com/pimcore/pimcore/issues/17669
Follow-up #17221

### CODE REVIEW

**Comment by mattamon** [comment]
@kingjia90 can you please check the sonarcloud issues and maybe resolve them?

File: models/Site.php [hunk 1]
```diff
@@ -230,7 +230,9 @@ public function setDomains(array|string $domains): static
             $domains = Serialize::unserialize($domains);
         }
         array_map(static function ($domain) {
-            if (!filter_var($domain, FILTER_VALIDATE_DOMAIN, FILTER_FLAG_HOSTNAME)) {
+            //replace all wildcars with a placeholder dummy string
```
Comment (mattamon) [inline models/Site.php]:
```suggestion
            //replace all wildcards with a placeholder dummy string
```

**Comment by kingjia90** [comment]
> @kingjia90 can you please check the sonarcloud issues and maybe resolve them?

resolved, could this PR be merged then? TIA

**Review by mattamon — APPROVED** [review]
LGTM

**Pull request merged**